lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,22 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_data = @data[:client_cert] || @data[:certificate_path]
-      private_key_data = @data[:client_key] || @data[:private_key_path]
-      private_key_pass = @data[:client_key_pass] || @data[:private_key_pass]
+      client_cert_data = if cert_data = @data[:client_cert_data]
+                           cert_data
+                         elsif path = (@data[:client_cert] || @data[:certificate_path])
+                           File.read path
+                         end
+
+      private_key_data = if key_data = @data[:private_key_data]
```

Comment:
Not exactly, I think I was unclear. I think for backwards compatibility it would be good to keep the options, I was just saying the precedence should be a little different. I think it should look something like this (note I'm not writing good code, just trying to show precedence):

```ruby
client_cert_data = @data[:client_cert_data] || File.read(@data[:client_key] || @data[:certificate_path))

client_key_data = @data[:client_key_data] || File.read(@data[:client_key] || @data[:private_key_path])
```

So I think what you have for client_cert_data is pretty much fine as is, it just shuffles around what the name is for the other thing (and the precedence). Does that help clarify?